packages/client/src/components/datasets/CreateForm.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import { Tag, Tooltip, Icon, Button, Input, Modal } from 'antd';
+import { Dataset } from './common';
+
+interface Props {
+  dataset?: Pick<Dataset, 'id' | 'name' | 'tags'>;
+  visible: boolean;
+  onClose: () => void;
+  onSubmit: (data: { id: string; name: string; tags: string[] }) => void;
+}
+
+export function DatasetCreateForm({
+  dataset,
+  visible,
+  onClose,
+  onSubmit,
+}: Props) {
+  const [id, setId] = React.useState('');
+  const [name, setName] = React.useState('');
+  const [tags, setTags] = React.useState([]);
+  const [inputVisible, setInputVisible] = React.useState(false);
+  const [inputValue, setInputValue] = React.useState('');
+  const inputRef = React.useRef(null);
+
+  React.useEffect(() => {
+    if (dataset?.id) {
+      setId(dataset.id);
+      setName(dataset.name);
+      setTags(dataset.tags);
+    }
+  }, [dataset]);
+
+  React.useEffect(() => {
+    if (inputVisible) {
+      inputRef.current?.input.focus();
+    }
+  }, [inputVisible]);
+
+  function genId(value: string) {
```

Comment:
This is an independent function, you can move it outside the function component.